src/factory.ts

```diff
@@ -37,22 +40,91 @@ function runCommand(
 
 function isGitHubRelease(
   command: string,
-  options: GitHubReleaseOptions | ReleasePROptions
-): options is GitHubReleaseOptions {
+  options: GitHubReleaseFactoryOptions | ReleasePRFactoryOptions
+): options is GitHubReleaseFactoryOptions {
   return command === 'github-release' && typeof options === 'object';
 }
 
 function run(runnable: ReleasePR | GitHubRelease) {
   return runnable.run();
 }
 
-function githubRelease(options: GitHubReleaseOptions): GitHubRelease {
-  return new GitHubRelease(options);
+function githubRelease(options: GitHubReleaseFactoryOptions): GitHubRelease {
+  const {
+    repoUrl,
+    defaultBranch,
+    token,
+    apiUrl,
+    octokitAPIs,
+    releaseType,
+    path,
+    packageName,
+    bumpMinorPreMajor,
+    releaseAs,
+    snapshot,
+    monorepoTags,
+    fork,
+    changelogSections,
+    lastPackageVersion,
+    versionFile,
+    ...remaining
```

Comment:
I seem to be missing where `remaining` is defined?

Reply:
I can't hyperlink to the exact location but search for the string "Rest in Object Destructuring " in https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Operators/Destructuring_assignment

in this case `remaining` is whatever is left of the `options` parameter that wasn't pulled out in the destructuring.

Reply:
neat, haven't used that pattern.